Marlin/stepper.h

```diff
@@ -105,8 +105,9 @@ class Stepper {
     static volatile uint32_t step_events_completed; // The number of step events executed in the current block
 
     #if ENABLED(ADVANCE) || ENABLED(LIN_ADVANCE)
-      static uint8_t old_OCR0A;
-      static volatile uint8_t eISR_Rate;
+      static volatile uint16_t nextMainISRrun;
+      static volatile uint16_t nextAdvanceISRrun;
+      static volatile uint16_t eISR_Rate;
```

Comment:
As long as these variables aren't accessed outside the ISR (and inside `CRITICAL_SECTION`) , they don't need to be declared '`volatile`' here. Removing `volatile` will save a few bytes in compilation because they're read more than once in ISRhandler itself. To ensure that they're safe for the single task, put them in the `private` data section.